test/SDK/MonitoringApplianceEmulatorTests/ViewModels/MetricChartPropertyControlViewModelTests.cs

```diff
@@ -0,0 +1,137 @@
+            MetricQueryResult metricQueryResult = new MetricQueryResult("name", "unit", new List<MetricTimeSeries>()
+            {
+                new MetricTimeSeries(
+                    this.timestamps.Zip(this.values, (t, v) => new MetricValues(t, v, v, v, v, 1)).ToList(),
+                    new List<KeyValuePair<string, string>>() { })
+            });
+
+            // Mock metric client result
+            this.metricClientMock = new Mock<IMetricClient>();
+            this.metricClientMock
+                .Setup(x => x.GetResourceMetricsAsync(this.resource.ToResourceId(), It.IsAny<QueryParameters>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync((string resourceId, QueryParameters queryParameters, CancellationToken ct) =>
+                {
+                    // Verify the query parameters
+                    Assert.AreEqual(Aggregation.Total, queryParameters.Aggregations.Single());
+                    Assert.AreEqual(TimeSpan.FromMinutes(5), queryParameters.Interval);
+                    Assert.AreEqual(new DateTime(2019, 4, 1, 10, 0, 0), queryParameters.StartTime);
+                    Assert.AreEqual(new DateTime(2019, 4, 1, 12, 0, 0), queryParameters.EndTime);
+                    Assert.AreEqual("metric1", queryParameters.MetricNames.Single());
+                    Assert.AreEqual("nameSpace", queryParameters.MetricNamespace);
+
+                    return new[] { metricQueryResult };
+                });
+
+            this.analysisServicesFactoryMock = new Mock<IAnalysisServicesFactory>();
+            this.analysisServicesFactoryMock
+                .Setup(x => x.CreateMetricClientAsync(this.resource.SubscriptionId, It.IsAny<CancellationToken>()))
+                .ReturnsAsync(this.metricClientMock.Object);
+
+            this.tracerMock = new Mock<ITracer>();
+        }
+
+        [TestMethod]
+        public async Task WhenCreatingViewModelForMetricChartWithDataThenThenItIsCreatedSuccessfully()
+        {
+            // Create view model
+            MetricChartPropertyControlViewModel viewModel = await this.CreateViewModel();
+
+            // Verify results
+            Assert.IsFalse(viewModel.ReadChartValuesTask.IsRunning);
+            ChartValues<DateTimePoint> points = viewModel.ReadChartValuesTask.Result;
+            Assert.IsNotNull(points);
+            Enumerable.SequenceEqual(this.timestamps, points.Select(p => p.DateTime));
+            Enumerable.SequenceEqual(this.values, points.Select(p => p.Value));
+        }
+
+        [TestMethod]
+        public async Task WhenCreatingViewModelForMetricChartWithInvalidStartTimeThenAnExceptionIsThrown()
+        {
+            this.property.StartTimeUtc = null;
+            MetricChartPropertyControlViewModel viewModel = await this.CreateViewModel();
+            Assert.IsNull(viewModel.ReadChartValuesTask.Result);
```

Comment:
>Assert [](start = 12, length = 6)

why you are not really validating that an exception was thrown and the thrown exception type?

Reply:
Because no exception is thrown, it is only traced. I'll add verification for the trace.

---
In reply to: [304760980](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/70#discussion_r304760980) [](ancestors = 304760980)